freeipa/src/main/java/com/sequenceiq/freeipa/sync/ProviderChecker.java

```diff
@@ -40,12 +43,22 @@
     @Value("${freeipa.autosync.update.status:true}")
     private boolean updateStatus;
 
+    public List<ProviderSyncResult> updateAndGetStatuses(Stack stack, boolean updateStatusFromFlow) {
+        Set<InstanceMetaData> notTerminatedForStack = stack.getAllInstanceMetaDataList().stream()
+                .filter(Predicate.not(InstanceMetaData::isTerminated))
+                .collect(Collectors.toSet());
+        Set<InstanceMetaData> checkableInstances = notTerminatedForStack.stream()
```

Comment:
It seems that these two collections could be combined into a single collection that is filtered twice.

Reply:
Thx! Updated it accordingly.